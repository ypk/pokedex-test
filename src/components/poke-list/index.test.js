import { render, screen, cleanup } from "@testing-library/react";
import PokeList from "./index";

afterEach(cleanup)

describe("Testing <PokeList />", () => {
    describe("renders without items", () => {
        it("should match snapshot", () => {
            const component = render(<PokeList listItems={[]} />);
            expect(component.asFragment()).toMatchSnapshot();
        });

        describe("rendering list functionality", () => {
            it("renders pokemon card", () => {
                render(<PokeList listItems={[{ "name": "pikachu", "height": 4, "weight": 60, "abilities": ["static", "lightning-rod"], "types": ["electric"] }]} />);
                const linkElement = screen.getByRole("link");
                expect(linkElement).toBeInTheDocument();
            });

            it("renders pokemon details correctly", () => {
                render(<PokeList listItems={[{ "name": "pikachu", "height": 4, "weight": 60, "abilities": ["static", "lightning-rod"], "types": ["electric"] }]} />);
                const linkElement = screen.getByRole("link");
                expect(linkElement).toHaveTextContent("Height4");
                expect(linkElement).toHaveTextContent("lightning-rod");
                expect(linkElement).toHaveAttribute("href", "https://pokemondb.net/pokedex/pikachu");
            });
        });

        describe("rendering empty list functionality", () => {
            it("renders pokemon not found message", () => {
                render(<PokeList listItems={[]} />);
                expect(screen.getByText("Not Found")).toBeInTheDocument();
                expect(screen.getByText("The pokemon you've searched is not found in the current set that's requested.")).toBeInTheDocument();
                expect(screen.getByText("Please refresh the page and try searching for a different pokemon.")).toBeInTheDocument();
            });

            it("renders refresh button", () => {
                render(<PokeList listItems={[]} />);
                const buttonElement = screen.getByRole("link");
                expect(buttonElement).toBeInTheDocument();
                expect(buttonElement).toHaveTextContent("Refresh Page");
                expect(buttonElement).toHaveAttribute("href", "/");
            });
        });
    });
});


