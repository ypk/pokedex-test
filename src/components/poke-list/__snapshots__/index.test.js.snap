// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <PokeList /> renders without items should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-gray-100 min-h-screen pt-18 py-6 md:pt-20 md:py-12 px-10"
  >
    <h2
      class="font-julius text-2xl font-bold my-4 mx-10"
    >
      Not Found
    </h2>
    <p
      class="text-xl my-4 mx-10"
    >
      The pokemon you've searched is not found in the current set that's requested.
    </p>
    <p
      class="text-xl my-4 mx-10"
    >
      Please refresh the page and try searching for a different pokemon.
    </p>
    <br />
    <a
      class="text-xl my-4 mx-10 ring ring-yellow-300 bg-yellow-300 hover:bg-yellow-400 focus:bg-yellow-400 py-3 px-4 rounded-xl"
      href="/"
    >
      Refresh Page
    </a>
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 md:gap-x-10 xl-grid-cols-5 gap-y-10 gap-x-6 "
    />
  </div>
</DocumentFragment>
`;
